gap/grahom.gi

```diff
@@ -652,6 +652,200 @@ function(src, ran, x, cols1, cols2)
+          Remove(currentpath);
+          if Size(currentpath) > 0 then
+            currentpath[Size(currentpath)] := currentpath[Size(currentpath)] + 1;
+          fi;
+        od;
+      else
+        Add(optiontree, newoptions);
+        Add(currentpath, 1);
+      fi;
+      #when we know we have a problem 
+      edgepath := List([1 .. Size(currentpath)],
+                       x -> optiontree[x][currentpath[x]]);
+      if not Size(Set(edgepath)) = Size(edgepath) then
+        Add(failedpairs, optiontree[1][1]);
+        break;
+      fi;
+    od;
+  od;
+
+  #checking backwards injective 
+  badedgepairs := failedpairs;
+  goodedgepairs := [];
+  while not badedgepairs = [] do
+    newpair := Remove(badedgepairs);
+    optiontree := [[newpair]];
+    currentpath := [1];
+    while not optiontree = [] do
+      currentpair := optiontree[Size(currentpath)]
+                               [currentpath[Size(currentpath)]];
+      newoptions := Cartesian(domedgesbyoutvertex[domedges[currentpair[1]][2]],
+                             domedgesbyoutvertex[domedges[currentpair[2]][2]]);
+      newoptions := Filtered(newoptions, x -> (x[1] ^ edgemap = x[2] ^ edgemap)
+                             and (x[1] < x[2]) and (not x in goodedgepairs));
+      if newoptions = [] then
+        currentpath[Size(currentpath)] := currentpath[Size(currentpath)] + 1;
+        while currentpath <> [] and currentpath[Size(currentpath)] >
+                                    Size(optiontree[Size(currentpath)]) do
+          goodedgepairs := Union(goodedgepairs, Remove(optiontree));
+          Remove(currentpath);
+          if Size(currentpath) > 0 then
+            currentpath[Size(currentpath)] := currentpath[Size(currentpath)]
+                                              + 1;
+          fi;
+        od;
+      else
+        Add(optiontree, newoptions);
+        Add(currentpath, 1);
+      fi;
+      #when we know we have a problem 
+      edgepath := List([1 .. Size(currentpath)],
+                       x -> optiontree[x][currentpath[x]]);
+      if not Size(Set(edgepath)) = Size(edgepath) then
+        return false;
+      fi;
+    od;
+  od;
+
+#checking surjective
+  loopable := Filtered(DigraphVertices(codomain),
+                       x -> IsReachable(codomain, x, x));
+  forwardcoverededges := Filtered([1 .. Size(codomedges)],
+                  x -> ForAny(loopable, y -> IsReachable(codomain,
+                                                         codomedges[x][2],
+                                                         y)));
+  coverededges := Filtered(forwardcoverededges,
+                  x -> ForAny(loopable, y -> IsReachable(codomain, y,
+                                                         codomedges[x][1])));
+
+  goodconfigurations := [];
+  configurationstocheck := List(coverededges,
+                                x -> [x, Filtered([1 .. Size(domedges)],
+                                                   y -> y ^ edgemap = x)]);
```

Comment:
I think I may be misunderstanding something here, but in the pdf you say 

> We start by checking which edges of the domain are involved in biinfinite paths as these edges can
> be ignored throughout

and I think these correspond to the `coverededges`, but here the `coverededges` come from the codomain, and they are not ignored?